### ISSUE TITLE
Undo registerDoFuture()

### ISSUE THREAD

**Issue by HenrikBengtsson** [djbetancourt-gh/funGp#15]
Analogously to #14, a user might have registered another **foreach** adaptor before calling your functions.  To undo `registerDoFuture()`, please consider updating your code to use:

```r
oldDoPar <- registerDoFuture()
on.exit(with(oldDoPar, foreach::setDoPar(fun=fun, data=data, info=info)), add = TRUE)
```

This is documented in Section 'For package developers' of [`?doFuture::registerDoFuture`](https://dofuture.futureverse.org/reference/registerDoFuture.html).

Now, for this to work, you also have to drop `registerDoRNG()`, which you can achieve by replacing `%dopar%` with `%dorng%`.  This is actually the best way to use **doRNG**, because it makes it explicit which `foreach()` statement relying on random number generation.  You can find documentation for this in Section 'Random Number Generation (RNG)' of [`?doFuture::doFuture`](https://dofuture.futureverse.org/reference/doFuture.html).

With the above + #14, you'll know that your functions will not interfer with user's pre-existing future or foreach settings.

**Comment by djbetancourt-gh** [comment]
Thank you, I was waiting this new version to take into account your useful suggestions. I've already taken into account the other two.

I'll have to put this one on hold while renozao
/doRNG#23 gets resolved. Hope it will be resolved on time!

**Comment by djbetancourt-gh** [comment]
Resolved in 70ae9f05801b43f4449b21bf6232b323561dc0f7
